x-pack/test/api_integration/deployment_agnostic/apis/observability/apm/service_overview/instance_details.spec.ts

```diff
@@ -0,0 +1,181 @@
+              query: {
+                start,
+                end,
+              },
+            },
+          });
+
+          expect(response.status).to.be(200);
+          expect(response.body).to.eql({});
+        });
+      });
+
+      // FLAKY: https://github.com/elastic/kibana/issues/177494
+      describe('when data is loaded', () => {
+        const range = timerange(new Date(start).getTime(), new Date(end).getTime());
+
+        const serviceInstance = apm
+          .service({ name: 'service1', environment: 'production', agentName: 'go' })
+          .instance('multiple-env-service-production');
+
+        const metricOnlyInstance = apm
+          .service({ name: 'service1', environment: 'production', agentName: 'java' })
+          .instance('multiple-env-service-production');
+
+        before(() => {
+          return client.index([
+            range
+              .interval('1s')
+              .rate(4)
+              .generator((timestamp) =>
+                serviceInstance
+                  .transaction({ transactionName: 'GET /api' })
+                  .timestamp(timestamp)
+                  .duration(1000)
+                  .success()
+              ),
+            range
+              .interval('30s')
+              .rate(1)
+              .generator((timestamp) =>
+                metricOnlyInstance
+                  .containerId('123')
+                  .podId('234')
+                  .appMetrics({
+                    'system.memory.actual.free': 1,
+                    'system.cpu.total.norm.pct': 1,
+                    'system.memory.total': 1,
+                    'system.process.cpu.total.norm.pct': 1,
+                  })
+                  .timestamp(timestamp)
+              ),
+          ]);
+        });
+
+        after(async () => {
+          await client.clean();
+        });
+
+        describe('fetch instance details', () => {
+          let response: {
+            status: number;
+            body: ServiceOverviewInstanceDetails;
+          };
+
+          let serviceNodeIds: string[];
+
+          before(async () => {
+            serviceNodeIds = await getServiceNodeIds({
+              apmApiClient,
+              start,
+              end,
+              serviceName: 'service1',
+            });
+
+            response = await apmApiClient.readUser({
+              endpoint:
+                'GET /internal/apm/services/{serviceName}/service_overview_instances/details/{serviceNodeName}',
+              params: {
+                path: { serviceName: 'service1', serviceNodeName: serviceNodeIds[0] },
+                query: {
+                  start,
+                  end,
+                },
+              },
+            });
+          });
+
+          it('returns the instance details', () => {
+            expect(response.body).to.not.eql({});
+          });
+
+          it('return the correct data', () => {
+            expect(omit(response.body, '@timestamp')).to.eql({
```

Comment:
is this replacing the snapshot?

Reply:
Yes, as this avoids the mismatched title issue with the snapshot testing. `expectSnapshot` still creates the snapshot files even though you specify ".toMatchInline".